File: fc-chat/demo/part3/ch3/05-1-message-system-add-channel/src/main/java/net/prostars/messagesystem/entity/UserConnectionId.java
package net.prostars.messagesystem.entity;

import java.io.Serializable;
import java.util.Objects;

public class UserConnectionId implements Serializable {

  private Long partnerAUserId;
  private Long partnerBUserId;

  public UserConnectionId() {}

  public UserConnectionId(Long partnerAUserId, Long partnerBUserId) {
    this.partnerAUserId = partnerAUserId;
    this.partnerBUserId = partnerBUserId;
  }

  public Long getPartnerAUserId() {
    return partnerAUserId;
  }

  public Long getPartnerBUserId() {
    return partnerBUserId;
  }

  @Override
  public boolean equals(Object o) {
    if (o == null || getClass() != o.getClass()) return false;
    UserConnectionId that = (UserConnectionId) o;
    return Objects.equals(partnerAUserId, that.partnerAUserId)
        && Objects.equals(partnerBUserId, that.partnerBUserId);
  }

  @Override
  public int hashCode() {
    return Objects.hash(partnerAUserId, partnerBUserId);
  }

  @Override
  public String toString() {
    return "UserConnectionId{partnerAUserId=%d, partnerBUserId=%d}"
        .formatted(partnerAUserId, partnerBUserId);
  }
}

File: fc-chat/demo/part3/with-push/src/main/kotlin/com/yscorp/withpush/messagesystem/entity/UserConnectionId.kt
package com.yscorp.withpush.messagesystem.entity

import java.io.Serializable
import java.util.*

class UserConnectionId(
    val partnerAUserId: Long,
    val partnerBUserId: Long,
) : Serializable {

    override fun equals(o: Any?): Boolean {
        if (o == null || javaClass != o.javaClass) return false
        val that = o as UserConnectionId
        return partnerAUserId == that.partnerAUserId
            && partnerBUserId == that.partnerBUserId
    }

    override fun hashCode(): Int {
        return Objects.hash(partnerAUserId, partnerBUserId)
    }

    override fun toString(): String {
        return "UserConnectionId{partnerAUserId=%d, partnerBUserId=%d}"
            .formatted(partnerAUserId, partnerBUserId)
    }
}
